### PR TITLE
[Minor][C++] Revise the unsupported data type error msg to give more information

### PR DESCRIPTION
**Pull request by acezen**
## Proposed changes
as title

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [ ] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: cpp/src/data_type.cc [hunk 1]
```diff
@@ -42,7 +42,8 @@ std::shared_ptr<arrow::DataType> DataType::DataTypeToArrowDataType(
   case Type::LIST:
     return arrow::list(DataTypeToArrowDataType(type->child_));
   default:
-    throw std::runtime_error("Unsupported data type");
+    std::string msg = "The data type " + type->ToTypeName() +
```
Comment (Thespica) [inline cpp/src/data_type.cc:45]: I wonder how the default case return or throw, because it doesn't `return` or `throw` directly.

Comment (lixueclaire) [reply cpp/src/data_type.cc:45]: throw std::runtime_error(msg);

Comment (acezen) [reply cpp/src/data_type.cc:45]: Thanks for pointing out, fixed.

**Review by lixueclaire — APPROVED** [review]
LGTM

**Pull request merged**